internal/archiver/archiver.go

```diff
@@ -388,19 +389,33 @@ func (arch *Archiver) Save(ctx context.Context, snPath, target string, previous
 
 		// use previous list of blobs if the file hasn't changed
 		if previous != nil && !fileChanged(fi, previous, arch.IgnoreInode) {
-			debug.Log("%v hasn't changed, using old list of blobs", target)
-			arch.CompleteItem(snPath, previous, previous, ItemStats{}, time.Since(start))
-			arch.CompleteBlob(snPath, previous.Size)
-			fn.node, err = arch.nodeFromFileInfo(target, fi)
-			if err != nil {
-				return FutureNode{}, false, err
+			contentsOK := true
+			// check if all blobs are contained in index
+			for _, id := range previous.Content {
+				if !arch.Repo.Index().Has(id, restic.DataBlob) {
+					contentsOK = false
+				}
```

Comment:
Maybe make this a function `allBlobsPresent`? This function is awfully long already.

Reply:
Thanks for the proposal! I changed this.